### PR TITLE
music fade in and example

### PR DESCRIPTION
**Pull request by mcpalmer1980**
This is my first time creating a pull request.

Added a fade_ms argument to the music.play() method and described it in music.rst.

Added a hopefully clean and well documented example of using it, along with set_endevent and other pygame/python features that might help new users.

### CODE REVIEW

**Comment by illume** [comment]
Awesome :) Thanks. 🎈 🎉

**Comment by mcpalmer1980** [comment]
As far as I can tell the python alerts are fixed and the travis-ci failure is not a problem with my code. I would guess it a fluke or a problem with pygame2/sdl2 on osx. Not sure what to do next.

File: examples/music_drop_fade.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    clock = pg.time.Clock()
+    pg.scrap.init()
+    pg.SCRAP_TEXT = pg.scrap.get_types()[0]  # TODO remove when  scrap module is fixed
+    clipped = pg.scrap.get(pg.SCRAP_TEXT)  # store the current text in the clipboard
+
+    # add the command line arguments to the file_list
+    for arg in sys.argv[1:]:
+        add_file(arg)
+    play_file("house_lo.ogg")  # play default music included with pygame
+    play_file("crap.ogg")  # TODO remove this test
```
Comment (illume) [inline examples/music_drop_fade.py]: 💩 hehe

File: examples/music_drop_fade.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    """
+    screen = pg.display.get_surface()
+    surf = font.render(text, 1, (255, 255, 255))
+    y += surf.get_height()
+    x = (screen.get_width() - surf.get_width()) / 2
+    screen.blit(surf, (x, y))
+    return y
+
+
+MUSIC_DONE = pg.USEREVENT + 1  # create a user event called MUSIC_DOWN in the first slot
```
Comment (illume) [inline examples/music_drop_fade.py]: We recently added: `pygame.event.custom_type()` https://www.pygame.org/docs/ref/event.html#pygame.event.custom_type

```python
MUSIC_DONE = pg.event.custom_type()
```

**Comment by illume** [comment]
Looks great. I made two small comments. I guess just the 'crap.ogg' file should be removed (if it stops it from running?).

**Comment by mcpalmer1980** [comment]
It was meant to be removed, but the 'crap.ogg' shouldn't stop it from running. It was actually alien1.gif renamed so to test a non audio file with an accepted extension, which prints "Not an Ogg Vorbis audio stream'. It should just say 'file not found.'

Maybe the line should stay? Accidental toilet humor never hurt anyone did it?

**Comment by mcpalmer1980** [comment]
Still not sure about the Travis CI error. I don't remember it ocurring last time. Do you think it's because of the missing referenced file? The details don't seem to suggest this to me.

**Comment by mcpalmer1980** [comment]
I think this is done now.

**Pull request merged**